klaus/contrib/auto_klaus.py

```diff
@@ -0,0 +1,103 @@
+"""
+Alternative take on the "automatically discovered repositories" concept
+that requires no threads, polling or inotify. Instead the filesystem is
+consulted whenever a repository name is looked up.
+
+Since os.path.exists() and os.listdir() are fairly quick filesystem
+operations, performance should be good for small to medium sites.
+FancyRepo() objects are cached.
+
+Repositories are identified by the existence of a
+
+    <reponame>/git-daemon-export-ok
+
+file (for compatibility with gitweb).
+
+Example usage:
+
+    from klaus.contrib.auto_klaus import Klaus, SlashDynamicRepos
```

Comment:
Maybe we can find a more descriptive name than `auto_klaus`

Reply:
Yeah this was just the first name I came up with as I was throwing this together for my own server, heh. I'm open to suggestions. The two hardest problems in computer science: naming things, cache invalidation and off-by-one errors. 😉

Reply:
Renamed it to `wsgi_autodetecting.py`. Still a bit generic but hopefully more descriptive.